app/models/article.rb

```diff
@@ -613,6 +614,17 @@ def skip_indexing?
 
   private
 
+  def collection_cleanup
+    # Should only check to cleanup if Article was removed from collection
+    return unless saved_change_to_collection_id? && collection_id.nil?
```

Comment:
The code works when tested locally, but this method/check is always returning `false` inside tests. I found [this SO answer](https://stackoverflow.com/a/60437898/3462026) that mentions `saved_change_to_collection_id` or `saved_change_to_attribute?(:collection_id)` won't work correctly on `after_save` or `after_update`, so `after_commit` needs to be used due to them being wrapped in a transaction.

I attempted all (strongly believe this but might missed one?) permutations of these options, but without any luck. My guess is that the fact the test is running inside a transaction itself (Rspec does this) makes this always return `false`. Happy to push those changes here if anyone feels like that would be better.

I also attempted to move this test to a system (capybara) spec but also without any luck. Any ideas on how to test this would be very welcome!

Reply:
Maybe you already tried this, but do you have the same result if you move the check into an `if` proc on the `after_save` itself? 

```ruby
after_save :collection_cleanup, if: proc { |article| article.saved_change_to_collection_id && article.collection_id.nil? }
```

Reply:
Thank's for the suggestion @jaw6! 🙏🏼 

Mac helped me sort out the failing spec which _seemed to be_ caused by a transient property in article factory ([thread here](https://forem-team.slack.com/archives/C7GE84DEJ/p1663195554524339?thread_ts=1663181073.686099&cid=C7GE84DEJ)). Tests working well now 🙌🏼